Source/BugsnagThread.m

```diff
@@ -20,11 +20,43 @@ @interface BugsnagThread ()
 @end
 
 @interface BugsnagStacktrace ()
++ (instancetype)stacktraceFromJson:(NSDictionary *)json;
 @property NSMutableArray<BugsnagStackframe *> *trace;
 @end
 
 @implementation BugsnagThread
 
++ (instancetype)threadFromJson:(NSDictionary *)json {
+    if (json == nil) {
+        return nil;
+    }
+    NSString *type = json[@"type"];
+    BSGThreadType threadType = [@"cocoa" isEqualToString:type] ? BSGThreadTypeCocoa : BSGThreadTypeReactNativeJs;
```

Comment:
Although relatively simple, it would be good to see the serialisation and parsing of thread type factored out into separate functions as per `BSGParseErrorType` on #588 to avoid repetition of the logic buried down this far.

Reply:
Factored out to separate functions.